djangofromscratch/JWTLogin/models.py

```diff
@@ -33,3 +80,10 @@ def get_absolute_url(self) -> str:
 
         """
         return reverse("users:detail", kwargs={"pk": self.id})
+
+    def save(self, *args, **kwargs):
+        try:
+            self.full_clean()
+            super(User, self).save(*args, **kwargs)
+        except IntegrityError:
+            raise ValidationError("error message")
```

Comment:
Shouldn't we need to return the right error message for better visibility

Reply:
Yes, I will correct it in new PR